aristanetworks/ansible-avd
# Fix(eos_validate_state): ANTA Remove the use of the eos_design's "type" variable

## Change Summary

Remove the use of `type: l3leaf` when testing. This is an eos_designs variable that can be customized and we cannot rely on this. Custom node types with VTEP role will now be tested properly.

## Related Issue(s)

Fixes #1047 

## Component(s) name

`arista.avd.eos_designs` **ANTA Mode**

## Proposed changes
- Added a method `is_vtep` to determine if a host is a VTEP or not based on the presence of a VXLAN interface.

## How to test
- Create a custom node type in eos_designs, rebuild your configs, run eos_validate_state in ANTA mode (see below) and make sure tests are added properly to the catalog of your custom nodes.
- `ansible-playbook playbooks/fabric-validate.yml -e use_anta=true -e save_catalog=true`

## Checklist

### User Checklist


- N/A

### Repository Checklist
